alert-common/src/main/java/com/synopsys/integration/alert/common/email/EmailMessagingService.java

```diff
@@ -167,49 +168,54 @@ private Session createMailSession(EmailProperties emailProperties) {
+        } catch (MessagingException e) {
+            String errorMessage = "Could not setup the email transport: " + e.getMessage();
+            logger.error(errorMessage);
+            throw new AlertException(errorMessage, e);
         }
         if (!errorMessages.isEmpty()) {
-            String joinedErrorMessages = StringUtils.join(errorMessages, ", ");
-            String errorMessage;
-            if (invalidRecipients.isEmpty()) {
-                errorMessage = "Errors sending emails. " + joinedErrorMessages;
-            } else {
-                errorMessage = String.format("Error sending emails to the following recipients: %s. %s.", invalidRecipients, joinedErrorMessages);
-            }
-            throw new AlertException(errorMessage);
+            String joinedErrorMessages = StringUtils.join(errorMessages, System.lineSeparator());
+            logger.error(joinedErrorMessages);
+            throw new AlertException(joinedErrorMessages);
         }
     }
 
-    private void sendAuthenticated(EmailProperties emailProperties, Message message, Session session) throws MessagingException {
+    private Transport getAndConnectTransport(EmailProperties emailProperties, Session session) throws MessagingException {
         String host = emailProperties.getJavamailOption(EmailPropertyKeys.JAVAMAIL_HOST_KEY);
         int port = NumberUtils.toInt(emailProperties.getJavamailOption(EmailPropertyKeys.JAVAMAIL_PORT_KEY));
         String username = emailProperties.getJavamailOption(EmailPropertyKeys.JAVAMAIL_USER_KEY);
         String password = emailProperties.getJavamailOption(EmailPropertyKeys.JAVAMAIL_PASSWORD_KEY);
 
-        Transport transport = session.getTransport("smtp");
-        try {
+        Transport transport = session.getTransport();
+        if (Boolean.valueOf(emailProperties.getJavamailOption(EmailPropertyKeys.JAVAMAIL_AUTH_KEY))) {
             transport.connect(host, port, username, password);
-            transport.sendMessage(message, message.getAllRecipients());
-        } finally {
-            transport.close();
+        } else {
+            transport.connect();
+        }
+        return transport;
+    }
+
+    private Set<String> sendMessage(Transport transport, Message message) {
+        Set<String> errorMessages = new HashSet<>();
+        Address[] recipients = null;
+        try {
+            recipients = message.getAllRecipients();
+            transport.sendMessage(message, recipients);
+        } catch (MessagingException e) {
+            Set<String> recipientAddresses = Collections.emptySet();
+            if (recipients != null) {
+                recipientAddresses = Stream.of(recipients).map(Address::toString).collect(Collectors.toSet());
+            }
+            String error = String.format("Could not send this email to the following recipients: %s. Reason: %s", recipientAddresses, e.getMessage());
+            errorMessages.add(error);
         }
+        return errorMessages;
```

Comment:
Is this Set ever going to have more than one element in it? Should we just return an Optional\<String\>?